gcloud/test__helpers.py

```diff
@@ -184,9 +186,11 @@ class Test__get_default_service_project_id(unittest2.TestCase):
     config_file = 'config_default'
 
     def setUp(self):
-        import tempfile
         import os
-        self.temp_config_path = tempfile.gettempdir()
+        from gcloud._testing import _NamedTemporaryFile
+
+        with _NamedTemporaryFile() as temp:
+            self.temp_config_path = os.path.dirname(temp.name)
 
```

Comment:
I think because I was just getting the dirname, it for some reason was happy to write to that location(found through `mkstemp`) instead of what I was `tempfile.NamedTemporaryFile` before.